zproject/backends.py

```diff
@@ -1430,11 +1430,16 @@ def social_associate_user_helper(
     full_name = kwargs["details"].get("fullname")
     first_name = kwargs["details"].get("first_name")
     last_name = kwargs["details"].get("last_name")
-    if all(name is None for name in [full_name, first_name, last_name]) and backend.name != "apple":
-        # Apple authentication provides the user's name only the very first time a user tries to log in.
+    if all(name is None for name in [full_name, first_name, last_name]) and backend.name not in [
+        "apple",
+        "saml",
+    ]:
```

Comment:
We could be a bit more fancy here and call a method e.g. `backend.verify_names(...)` which would raise AssertionErrors for most social backends and be overriden to pass for apple and saml backends, but it doesn't seem necessary for now, while the exceptions list only has 2 elements

Reply:
Yeah, that seems like a fine long-term plan, but we can leave it for future work.